src/main/scala/sparkz/core/network/peer/PeerManager.scala

```diff
@@ -3,23 +3,47 @@ package sparkz.core.network.peer
 import akka.actor.{Actor, ActorRef, ActorSystem, Props}
 import sparkz.core.app.SparkzContext
 import sparkz.core.network._
+import sparkz.core.network.peer.PeerDatabase.{PeerConfidence, PeerDatabaseValue}
+import sparkz.core.persistence.ScheduledActor.ScheduledActorConfig
+import sparkz.core.persistence.{PersistablePeerDatabase, ScheduledStoragePersister, StoragePersister, StoragePersisterContainer}
 import sparkz.core.settings.SparkzSettings
 import sparkz.core.utils.NetworkUtils
 import sparkz.util.SparkzLogging
-import sparkz.core.network.peer.PeerDatabase.{PeerConfidence, PeerDatabaseValue}
 
 import java.net.{InetAddress, InetSocketAddress}
-import scala.util.Random
+import java.nio.file.Files
 import java.security.SecureRandom
+import scala.concurrent.ExecutionContext
+import scala.util.Random
 
 /**
   * Peer manager takes care of peers connected and in process, and also chooses a random peer to connect
   * Must be singleton
   */
-class PeerManager(settings: SparkzSettings, sparkzContext: SparkzContext, peerDatabase: PeerDatabase) extends Actor with SparkzLogging {
+class PeerManager(
+                   settings: SparkzSettings,
+                   sparkzContext: SparkzContext,
+                   peerDatabase: PersistablePeerDatabase)(implicit ec: ExecutionContext)
+  extends Actor with SparkzLogging {
 
   import PeerManager.ReceivableMessages._
 
+  private val initializeStorageDirectory: () => Unit = () => Files.createDirectories(settings.dataDir.toPath)
+  private val storagePersisterContainer = new StoragePersisterContainer(
+    getAsScheduled,
+    initializeStorageDirectory
+  )
+  storagePersisterContainer.restoreAllStorages()
+
+  private def getAsScheduled: Seq[ScheduledStoragePersister] = {
+    val storagesToBackUp: Seq[StoragePersister[_]] = peerDatabase.storagesToPersist()
```

Comment:
one more comment about the name `persister`. Here `storagesToBackUp` is used - seems to be a good name